tests/test_select.py

```diff
@@ -45,75 +45,82 @@ async def send(
         ch1: Sender[int], ch2: Sender[int], ch3: Sender[int], queue: AsyncIterable
     ) -> None:
         for ctr in range(5):
-            await ch1.send(ctr + 1)
-            await ch2.send(ctr + 101)
-            await ch3.send(ctr + 201)
-            await queue.add(ctr + 301)
+            print(f"SEND: ch1: {ctr+101}")
```

Comment:
Also originally, before I introduced these `print()`s, I got the processing of `ch1` at the end, not the beginning:

```
>       assert results == expected_results
E       assert [101, 201, 401, 202, 402, 203, 403, 204, 404, 205, 405, 1000, -4, -2, 102, 103, 104, 105, -1] == [101, 201, 401, 102, 202, 402, 103, 203, 403, 104, 204, 404, 105, 205, 405, -1, 1000, -4, -2]
E         At index 3 diff: 202 != 102
E         Full diff:
E           [
E            101,
E            201,
E            401,
E         -  102,
E            202,
E            402,
E         -  103,
E            203,
E            403,
E         -  104,
E            204,
E            404,
E         -  105,
E            205,
E            405,
E         -  -1,
E            1000,
E            -4,
E            -2,
E         +  102,
E         +  103,
E         +  104,
E         +  105,
E         +  -1,
E           ]
```

This is reproduceable for me, if I remove the prints I still get this.